### PR TITLE
[fate#317775]: AutoYaST tests for btrfs non-default subvolumes structure

### PR DESCRIPTION
**Pull request by rwx788**
Tests for autoyast clone for btrfs. Tests are using same structure as for ext4, but with additional checks of functionality, namely:
1) btrfs cloned system matches autoinst profile configuration
2) autoinst.xml is generated according to the system btrfs configuration

For execution following job properties have to be set:
AUTOYAST 	autoyast/autoyast_btrfs.xml
FILESYSTEM 	btrfs

Additionally, worker should have XML::LibXML perl module installed.

### CODE REVIEW

**Comment by okurz** [comment]
Hi @rwx788 you can write `[WIP]` and no one will merge this PR but it will still invite for feedback.

**Comment by okurz** [comment]
make sure to add your perl module dependencies into the cpanfile

File: tests/autoyast/verify_btrfs.pm [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+use base 'basetest';
```
Comment (okurz) [inline tests/autoyast/verify_btrfs.pm]: Please add a summary and maintainer line as we have it in other files.

File: tests/autoyast/verify_btrfs.pm [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2014-2017 SUSE LLC
```
Comment (okurz) [inline tests/autoyast/verify_btrfs.pm]: Just use the years you have worked on this file. I assume just 2017 ;-)

File: tests/autoyast/verify_autoinst_btrfs.pm [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+sub verify_mount_opts {
+    my ( $mount_path, $mount_opts ) = @_;
+    my $nodeset =
+      $xpc->findnodes("//ns:partition[ns:mount=\"$mount_path\"]/ns:fstopt");
+
+    my @nodes = $nodeset->get_nodelist;
+    ## Verify that there is node found by xpath and it's single one
+    if ( scalar @nodes != 1 ) {
+        return
+"Generated autoinst.xml contains multiple partiotions with same mount path for"
```
Comment (okurz) [inline tests/autoyast/verify_autoinst_btrfs.pm]: s/partiotions/partitions/

File: tests/autoyast/verify_btrfs.pm (hunk 2 above)
Comment (rwx788) [inline tests/autoyast/verify_btrfs.pm]: Fixed.

File: tests/autoyast/verify_btrfs.pm (hunk 1 above)
Comment (rwx788) [inline tests/autoyast/verify_btrfs.pm]: Fixed.

File: tests/autoyast/verify_autoinst_btrfs.pm (hunk 3 above)
Comment (rwx788) [inline tests/autoyast/verify_autoinst_btrfs.pm]: Fixed.

**Comment by okurz** [comment]
there are a lot of unrelated commits in this PR. I don't know how you got them there. Please rebase to have only your own commits in the branch.

**Comment by okurz** [comment]
Please provide a link or screenshot to a local verification run.

**Comment by rwx788** [comment]
Local run results:
image

**Comment by okurz** [comment]
nice, thx. What about the rebase?

File: lib/main_common.pm [hunk 4]
```diff
@@ -157,8 +157,14 @@ sub load_autoyast_tests {
     else {
         loadtest("autoyast/repos");
         loadtest("autoyast/clone");
+        if (get_var('SP3ORLATER') && check_var('FILESYSTEM', 'btrfs')) {
+            loadtest "autoyast/verify_autoinst_btrfs";
+        }
         loadtest("autoyast/logs");
     }
+    if (get_var('SP3ORLATER') && check_var('FILESYSTEM', 'btrfs')) {
```
Comment (okurz) [inline lib/main_common.pm]: oops, I merged it now but I guess this is never true. I have never seen this variable "SP3ORLATER". Rather use `sle_version_at_least('12-SP3') || check_var('DISTRI', 'opensuse')`

Comment (rwx788) [inline lib/main_common.pm]: I found it in lib/registration.pm and in main.pm we have 
if (sle_version_at_least('12-SP3')) {
    set_var('SP3ORLATER', 1);
}

Should I change it in both places then?

Comment (okurz) [inline lib/main_common.pm]: as discussed in person, you were right, I was wrong. everything fine :-) Make sure the tests are triggered accordingly on the production instance

Comment (rwx788) [inline lib/main_common.pm]: Yep, tests are added and I keep an eye on execution results. Thanks!

**Comment by rwx788** [comment]
Run in production: https://openqa.suse.de/tests/955187

**Pull request merged**